sw/device/silicon_creator/lib/drivers/usb.h

```diff
@@ -0,0 +1,170 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_USB_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_USB_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/drivers/stdusb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Flags for endpoint configuration
+ */
+typedef enum usb_endpoint_type {
+  /** Endpoint is an IN endpoint */
+  kUsbEndpointTypeIn = 1,
+  /** Endpoint is a OUT endpoint */
+  kUsbEndpointTypeOut = 2,
+  /** Endpoint can accept SETUPDATA (e.g. a control endpoint) */
+  kUsbEndpointTypeSetup = 4,
+  /** Endpoint is a CONTROL endpoint */
+  kUsbEndpointTypeControl = 7,
+} usb_endpoint_type_t;
+
+/**
+ * Flags for managing usb transfers
+ */
+typedef enum usb_transfer_flags {
+  /** Transfer direction is IN (device to host) */
+  kUsbTransferFlagsIn = 0,
+  /** Transfer direction is OUT (host to device) */
+  kUsbTransferFlagsOut = 1,
+  /** Transfer direction mask */
+  kUsbTransferFlagsDirMask = 1,
+  /** Transfer is a control transfer: requires a zero-byte packet in the
+     opposite direction of the transfer to complete */
+  kUsbTransferFlagsControl = 2,
+  /** Transfer needs to terminate with a short or zero-byte packet. */
+  kUsbTransferFlagsShortIn = 4,
+
+  /**
+   * Indicates a SETUP_DATA packet; the data argument of the callback points to
+   * a usb_setup_data_t.
+   */
+  kUsbTransferFlagsSetupData = 0x2000,
+  /**
+   * Transfer is finished; the data argument of the callback points to a size_t
+   * indicating the number of bytes transferred.
+   */
+  kUsbTransferFlagsDone = 0x4000,
+  /** USB device was reset */
+  kUsbTransferFlagsReset = 0x8000,
+} usb_transfer_flags_t;
+
+/**
+ * Function pointer type for an endpoint handler.
+ *
+ * @param ctx A pointer to the context object supplied during endpoint
+ *            initialization.
+ * @param ep The endpoint number.
+ * @param flags The usb transfer flags for this callback.
+ * @param data A pointer to data relevant to this callback (see the flags).
+ */
+typedef void (*handler_t)(void *ctx, size_t ep, usb_transfer_flags_t flags,
+                          void *data);
```

Comment:
Endpoint IDs can fit in a uint8_t, including the direction. Did `size_t` have a specific motivation in mind?

Reply:
This is rust's influence.  `size_t` is the index type (just as `usize` is in rust).